product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/utils/XMLComparatorInterop.java

```diff
@@ -0,0 +1,212 @@
+import org.apache.axiom.om.OMNode;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.Iterator;
+import java.util.Vector;
+
+/**
+ * Util class for comparing two OMElements
+ */
+public class XMLComparatorInterop {
+
+    private static final Log log = LogFactory.getLog(XMLComparatorInterop.class);
+    //if namespaces needs to be ignored need to add to this list
+    private static Vector ignorableNamespaceList = new Vector();
+
+    /**
+     * Compares two OMElements. It compares attributes, namespaces, elements with trimmed values.
+     *
+     * @param elementOne first element to compare
+     * @param elementTwo element to compare with the first
+     * @return true if compares
+     */
+    public static boolean compare(OMElement elementOne, OMElement elementTwo) {
+
+        boolean status;
+        //ignore if the elements belong to any of the ignorable namespaces list
+
+        if (isIgnorable(elementOne) ||
+                isIgnorable(elementTwo)) {
+            return true;
+        }
+
+        if (elementOne == null && elementTwo == null) {
+            log.info("Both Elements are null.");
+            return true;
+        }
+        if (elementOne == null && elementTwo != null) {
+            return false;
+        }
+        if (elementOne != null && elementTwo == null) {
+            return false;
+        }
+
+        log.debug("Now Checking " + elementOne.getLocalName() + " and " +
+                elementTwo.getLocalName() +
+                "=============================");
+
+        log.debug("Comparing Element Names .......");
+        status = compare(elementOne.getLocalName(), elementTwo.getLocalName());
+        if (!status)
+            return false;
+
+        log.debug("Comparing Namespaces .........");
+        status = compare(elementOne.getNamespace(),
+                elementTwo.getNamespace());
+        if (!status)
+            return false;
+
+        log.debug("Comparing attributes .....");
+        status = compareAllAttributes(elementOne, elementTwo);
+        if (!status)
+            return false;
+
+        log.debug("Comparing texts .....");
+
+        /*
+         * Trimming the value of the XMLElement is not correct
+         * since this compare method cannot be used to compare
+         * element contents with trailing and leading whitespaces
+         * BUT for the practicalltiy of tests and to get the current
+         * tests working we have to trim() the contents
+         */
+        status = compare(
+                elementOne.getText().trim(),
+                elementTwo.getText().trim());
+        if (!status)
+            return false;
+
+        log.debug("Comparing Children ......");
+        status = compareAllChildren(elementOne, elementTwo);
+
+        return status;
+    }
+
+    private static boolean compareAllAttributes(OMElement elementOne,
+                                                OMElement elementTwo) {
```

Comment:
Can we merge this line with the previous?

Reply:
Done